threadpoolctl.py

```diff
@@ -510,3 +510,54 @@ def unregister(self):
         if self._original_limits is not None:
             for module in self._original_limits:
                 module['set_num_threads'](module['num_threads'])
+
+
+class safe_nested_parallelism:
+    """Context manager to ensure that the inner parallel loop run serially
+
+    This makes it possible to nest calls with libraries built with independent
+    threadpool runtimes without causing oversubscription.
+    """
+
+    def __init__(self, outer_module, outer_api, inner_module, inner_api):
```

Comment:
Yes I am not satisfied with this API either. Furthermore I don't see any way to make it work under Windows because the symbols are not bound tot the compiled python extensions themselves I and I am not sure how to trace how dynlib are linked to a specific Python extension.

Reply:
> Wouldn't it be simpler to implement an API with "limit all except one"?

I don't know how to do that either.